src/iuse.cpp

```diff
@@ -2122,9 +2122,25 @@ int iuse::water_purifier( player *p, item *it, bool, const tripoint & )
         return 0;
     }
 
-    p->moves -= to_moves<int>( 2_seconds );
+    int factor;
+    if( it->type->tool != NULL ) {
+        factor = it->type->tool->purification_factor;
+    } else if( it->type->comestible != NULL ) {
```

Comment:
```suggestion
    } else if( it->type->comestible != nullptr ) {
```

Reply:
clang-tidy requires using the C++ `nullptr` over `NULL`, though tbh in this code base the status quo would be
```cpp
if( it->type->comestible ) {
   ...
```